src/js/server/login.js

```diff
@@ -0,0 +1,145 @@
+"use strict";
+var fluid  = fluid || require("infusion");
+var gpii   = fluid.registerNamespace("gpii");
+
+fluid.registerNamespace("gpii.express.user.api.login.post.handler");
+
+var path      = require("path");
+var schemaDir = path.resolve(__dirname, "../../schemas");
+
+require("gpii-handlebars");
+require("gpii-json-schema");
+
+require("./lib/datasource");
+require("./lib/password");
+
+fluid.registerNamespace("gpii.express.user.api.login");
+
+gpii.express.user.api.login.post.handler.verifyPassword = function (that, response) {
+    if (that.request.body && that.request.body.username && that.request.body.password && response.salt && response.derived_key) {
+        var encodedPassword = gpii.express.user.password.encode(that.request.body.password, response.salt, response.iterations, response.keyLength, response.digest);
+        if (encodedPassword === response.derived_key) {
+            // Transform the raw response to ensure that nothing sensitive is exposed to the user
+            var user = fluid.model.transformWithRules(response, that.options.rules.user);
+            that.request.session[that.options.sessionKey] = user;
+            that.sendResponse(200, { ok: true, message: "You have successfully logged in.", user: user});
+            return;
+        }
+    }
+
+    that.sendResponse(401, { ok: false, message: "Invalid username or password."});
```

Comment:
Ah, I see that you have been using the "early return" pattern - an "else" here would be logically clearer